UPGRADE_NOTES.md

```diff
@@ -1,4 +1,79 @@
-### Upgrading from 0.14 to 0.15
+## Upgrading from 0.16 to 0.17
+
+- Ensure you've upgraded to crystal 0.30.0
+- Upgrade Lucky CLI (homebrew)
+
+```
+brew update
+brew upgrade crystal-lang # Make sure you're up-to-date. Requires 0.30.0
+brew upgrade lucky
+```
+
+- Upgrade Lucky CLI (Linux)
+
+- Update `.crystal-version` to `0.30.0`
+
+> Remove the existing Lucky binary and follow the Linux
+> instructions in this section
+> https://luckyframework.org/guides/getting-started/installing#on-linux
+
+- Update versions in `shard.yml`
+  - Lucky should be `~> 0.17`
+- Run `shards update`
+
+### General updates
+- Rename: Action rendering method `text` to `plain_text`.
+- Update: use of `number_to_currency` now returns a String instead of writing to the view directly.
+- Delete: `config/static_file_handler.cr`. The `Lucky::StaticFileHandler` no longer has config settings.
+- Add: a new `Lucky::LogHandler` configure to `config/logger.cr`.
+```
+Lucky::LogHandler.configure do |settings|
+  # Skip logging static assets in development
+  if Lucky::Env.development?
+    settings.skip_if = ->(context : HTTP::Server::Context) {
+      context.request.method.downcase == "get" &&
+      context.request.resource.starts_with?(/\/css\/|\/js\/|\/assets\//)
+    }
+  end
+end
+```
+
+### Database updates
+- Add: a new `AppDatabase` class in `config/database.cr` that inherits from `Avram::Database`. (e.g. `class AppDatabase < Avram::Database`)
+- Rename: `Avram::Repo.configure` to `AppDatabase.configure` in `config/database.cr`.
+- Add: `Avram.configure` block.
+```
+Avram.configure do |settings|
+  settings.database_to_migrate = AppDatabase
+  # this is moved from your old `Avram::Repo.configure` block.
+  settings.lazy_load_enabled = Lucky::Env.production?
+end
+```
+- Move: the `settings.lazy_load_enabled` from `AppDatabase.configure` to `Avram.configure` block.
```

Comment:
Not mentioned is that you should add `self.database` to the `BaseModel`

```diff
 abstract class BaseModel < Avram::Model
+  def self.database
+    AppDatabase
+  end
 end

```

Reply:
I found that `src/components/shared/field_errors.cr` got [an overhaul](https://github.com/luckyframework/lucky_cli/blob/37c40101a890a9c912168823e3f72f0bcd60beaa/src/browser_app_skeleton/src/components/shared/field_errors.cr) since I last looked at it which maybe we could tell people to optionally change.

Even if we don't, we need to say `Avram::FillableField` changed to be `Avram::PermittedAttribute`